src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.cs

```diff
@@ -372,15 +381,51 @@ public bool IsGeneric
             public bool IsAttribute
                 => (_properties & ItemPropertyKind.IsAttribute) != 0;
 
+            public bool IsEditorBrowsableStateAdvanced
+                => (_properties & ItemPropertyKind.IsEditorBrowsableStateAdvanced) != 0;
+
             public TypeImportCompletionItemInfo WithItem(CompletionItem item)
-                => new TypeImportCompletionItemInfo(item, IsPublic, IsGeneric, IsAttribute);
+                => new TypeImportCompletionItemInfo(item, IsPublic, IsGeneric, IsAttribute, IsEditorBrowsableStateAdvanced);
 
             [Flags]
             private enum ItemPropertyKind : byte
             {
                 IsPublic = 0x1,
                 IsGeneric = 0x2,
                 IsAttribute = 0x4,
+                IsEditorBrowsableStateAdvanced = 0x8,
+            }
+        }
+
+        // Things needed for determining whether a symbol is EditorBrowsable.
+        // Grouped together and reused within each compilation.
+        private class EditorBrowsableInfo
+        {
+            private Optional<IMethodSymbol?>? _editorBrowsableAttributeConstructor;
+            private ImmutableArray<IMethodSymbol>? _typeLibTypeAttributeConstructors;
+            private ImmutableArray<IMethodSymbol>? _typeLibFuncAttributeConstructors;
+            private ImmutableArray<IMethodSymbol>? _typeLibVarAttributeConstructors;
+
+            public Compilation Compilation { get; }
+
+            public Optional<IMethodSymbol?> EditorBrowsableAttributeConstructor
+                => _editorBrowsableAttributeConstructor ??= new(EditorBrowsableHelpers.GetSpecialEditorBrowsableAttributeConstructor(Compilation));
+
+            public ImmutableArray<IMethodSymbol> typeLibTypeAttributeConstructors
```

Comment:
```suggestion
            public ImmutableArray<IMethodSymbol> TypeLibTypeAttributeConstructors
```

(i'm amazed nothing flagged that).